### ISSUE TITLE
Files endpoint fails for hosted published workflow with test parameter json

### ISSUE THREAD

**Issue by coverbeck** [ga4gh/dockstore#2085]
## Bug Report

When viewing a published hosted workflow that has a test parameter JSON, there is an error fetching the version's files using the GA4GH API. If the hosted workflow has no JSON, then there is no error.

Interestingly, although I see the error in Chrome, I haven't found yet how the UI's behavior has changed. :(

This endpoint also fails in production, but the UI in production never invokes that API.

### Steps to reproduce the behaviour 

1. `curl 'https://staging.dockstore.org/api/api/ga4gh/v2/tools/%23workflow%2Fdockstore.org%2Fcoverbeck%2Fhelloworld/versions/4/CWL/files'` 
2. Yields `{"code":500,"message":"There was an error processing your request. It has been logged (ID 0ead45002d64e786)."}`

On the server side:

```
ERROR [2019-02-05 18:50:10,919] io.dropwizard.jersey.errors.LoggingExceptionMapper: Error handling a request: 9e9b2bcd57255637
! java.lang.IllegalArgumentException: 'other' is different type of Path
! at sun.nio.fs.UnixPath.relativize(UnixPath.java:416)
! at sun.nio.fs.UnixPath.relativize(UnixPath.java:43)
! at io.swagger.api.impl.ToolsApiServiceImpl.lambda$getToolFiles$9(ToolsApiServiceImpl.java:722)
! at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
! at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1380)
! at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
! at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
! at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
! at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
! at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
! at io.swagger.api.impl.ToolsApiServiceImpl.getToolFiles(ToolsApiServiceImpl.java:727)
! at io.swagger.api.impl.ToolsApiServiceImpl.toolsIdVersionsVersionIdTypeFilesGet(ToolsApiServiceImpl.java:652)
! at io.swagger.api.ToolsApi.toolsIdVersionsVersionIdTypeFilesGet(ToolsApi.java:201)
! at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source)
! at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
! at java.lang.reflect.Method.invoke(Method.java:498)
! at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
```

┆Issue is synchronized with this [Jira Bug](https://ucsc-cgl.atlassian.net/browse/DOCK-254)
┆Issue Type: Bug
┆Fix Versions: Dockstore 1.6
┆Issue Number: DOCK-254

**Comment by garyluu** [comment]
`This endpoint also fails in production, but the UI in production never invokes that API.`
`This endpoint also fails in staging, but the UI in production never invokes that API.`?

**Comment by garyluu** [comment]
The UI needs to call that endpoint in order to get the wget command in the launch tab.  The call is done so in develop and staging whereas production currently may end up with an incorrect command.

**Comment by coverbeck** [comment]
Curling the API fails on both production and staging.

I left out details on the UI behavior; when you go to the workflow's page in the UI, either on the public page or in My Workflows:

1. In production, the endpoint does not get invoked.
2. In staging, the UI invokes the endpoint and you can see the failure in Dev Tools.

**Comment by garyluu** [comment]
Andrew may have fixed this already https://github.com/ga4gh/dockstore/pull/2212

**Comment by garyluu** [comment]
https://github.com/ga4gh/dockstore/commit/d3bf79f463bde58b996e7947f8e4b2ab35483cf8 d3bf79f463bde58b996e7947f8e4b2ab35483cf8 to be reverted after validDescriptors is fixed.

**Comment by coverbeck** [comment]
Confirmed that it is fixed in 1.6.0-rc.0; curling the endpoint now returns a JSON object:

```
curl "https://staging.dockstore.org/api/api/ga4gh/v2/tools/%23workflow%2Fdockstore.org%2Fcoverbeck%2Fhelloworld/versions/3/CWL/files"
[{"file_type":"PRIMARY_DESCRIPTOR","path":"Dockstore.cwl"},{"file_type":"TEST_FILE","path":"test.cwl.json"}]
```
